File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/Java/MedicDeFamilie/src/familydoctor/repository/AbstractRepository.java
package familydoctor.repository;

import java.util.ArrayList;
import java.util.List;

public class AbstractRepository<E> {

    protected List<E> entities = new ArrayList<E>();

    public long size() {
        return entities.size();
    }

    public List<E> getAll() {
        return entities;
    }
}

File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/CSharp/FamilyDoctor/FamilyDoctor/Repository/AbstractRepository.cs
using System;
using System.Collections.Generic;

namespace FamilyDoctor.Repository
{
    public enum RepositoryFileFormat
    {
        XML,
        CSV
    }

    public class AbstractRepository<T>
    {
        protected List<T> entities = new List<T>();

        public List<T> GetAll()
        {
            return entities;
        }
    }
}